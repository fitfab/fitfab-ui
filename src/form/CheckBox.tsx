import React from 'react'
import styled from 'styled-components'

const CheckBoxView = styled.input`
    &[type='checkbox'] {
        display: none;
    }

    & + label {
        box-sizing: border-box;
        display: inline-block;
        height: 24px;
        line-height: 24px;
        min-width: auto;
        margin: 0 12px 0 40px;
        cursor: pointer;
        position: relative;
    }
    & + label:before {
        border: 1px solid #999;
        border-radius: 4px;
        display: block;
        content: '';
        height: 24px;
        width: 24px;
        line-height: 1.7;
        text-align: center;
        position: absolute;
        left: -40px;
    }
    &:checked + label:after {
        display: block;
        content: '';
        border-left: 2px solid green;
        border-bottom: 2px solid green;
        color: green;
        height: 4px;
        width: 12px;
        transform: rotate(-45deg);
        position: absolute;
        left: -34px;
        top: 8px;
    }
`

export interface CheckBoxProps {
    name?: string
    id?: string
    value?: string
    label?: string
}

export const CheckBox: React.FC<CheckBoxProps> = (props: CheckBoxProps) => {
    return (
        <React.Fragment>
            <CheckBoxView name={props.name} id={props.id} type="checkbox" />
            <label htmlFor={props.id}>{props.label}</label>
        </React.Fragment>
    )
}
