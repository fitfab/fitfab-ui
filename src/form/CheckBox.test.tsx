import { cleanup, render } from '@testing-library/react'
import React from 'react'
import { CheckBox } from './CheckBox'

afterEach(cleanup)

test('Render CheckBox', () => {
    const { container } = render(<CheckBox label="no" />)
    expect(container).toMatchSnapshot()
})
