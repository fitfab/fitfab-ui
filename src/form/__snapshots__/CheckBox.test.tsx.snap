// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render CheckBox 1`] = `
<div>
  <input
    class="sc-bdVaJa bfEAcK"
    type="checkbox"
  />
  <label>
    no
  </label>
</div>
`;
